api/v1alpha1/envoyproxy_types.go

```diff
@@ -97,6 +97,7 @@ type EnvoyProxyProvider struct {
 	// optional auxiliary control planes. Supported types are "Kubernetes".
 	//
 	// +unionDiscriminator
+	// +kubebuilder:validation:XValidation:message="type must be one of ['Kubernetes']",rule="self == 'Kubernetes'"
```

Comment:
is this needed ?
it already exists https://github.com/envoyproxy/gateway/blob/bdb0191dec04788a1b8e47b5a66f0cc99ef29afe/api/v1alpha1/shared_types.go#L36